src/test/util/blocktools_tests.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2019 The Unit-e developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/blocktools.h>
+
+#include <random.h>
+#include <uint256.h>
+
+#include <test/test_unite.h>
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(blocktools_tests)
+
+BOOST_AUTO_TEST_CASE(BLockIndexFake_MakeBlockIndex) {
+
+  blocktools::BlockIndexFake fake;
+
+  BOOST_CHECK_EQUAL(fake.block_indexes.size(), 0);
+
+  {
+    // create a block at height=0
+    const uint256 hash = GetRandHash();
+    CBlockIndex *block_index = fake.MakeBlockIndex(hash, nullptr);
+
+    BOOST_CHECK_EQUAL(fake.block_indexes.size(), 1);
+    BOOST_CHECK_EQUAL(&fake.block_indexes.begin()->second, block_index);
+    BOOST_CHECK_EQUAL(block_index->nHeight, 0);
+    BOOST_CHECK(block_index->pprev == nullptr);
+    BOOST_REQUIRE(block_index->phashBlock);
+    BOOST_CHECK_EQUAL(*block_index->phashBlock, hash);
+  }
+
+  {
+    // create a successor to the previously generated block
+    const uint256 hash = GetRandHash();
+    CBlockIndex *block_index = fake.MakeBlockIndex(hash, &fake.block_indexes.begin()->second);
+
+    BOOST_CHECK_EQUAL(fake.block_indexes.size(), 2);
+    BOOST_CHECK_EQUAL(block_index->nHeight, 1);
+    BOOST_CHECK(block_index->pprev);
+    BOOST_REQUIRE(block_index->phashBlock);
+    BOOST_CHECK_EQUAL(*block_index->phashBlock, hash);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(BlockIndexFake_Generate) {
```

Comment:
It would be nice to add a test for a non-existing `starting_point`.

Reply:
I just refactored this to accept a `CBlockIndex *` in response to a review comment.

This argument defaults to `nullptr` which generates a block at height 0, and this case is tested. In case one gives a block index which is not known by the fake map, a `BOOST_REQUIRE_MESSAGE` now fails.

I would not know how to write a check that asserts that a boost unit test requirement fails...